lib/pages/modlog_page.dart

```diff
@@ -0,0 +1,519 @@
+
+    InlineSpan user(UserSafe user) => TextSpan(
+          children: [
+            WidgetSpan(
+              child: Avatar(
+                url: user.avatar,
+                noBlank: true,
+                radius: 10,
+              ),
+            ),
+            TextSpan(
+              text: ' ${user.displayName}',
+              style: TextStyle(color: theme.accentColor),
+              recognizer: TapGestureRecognizer()
+                ..onTap = () => goToUser.byId(
+                      context,
+                      user.instanceHost,
+                      user.id,
+                    ),
+            ),
+          ],
+        );
+
+    InlineSpan community(CommunitySafe community) => TextSpan(
+          children: [
+            WidgetSpan(
+              child: Avatar(
+                url: community.icon,
+                noBlank: true,
+                radius: 10,
+              ),
+            ),
+            TextSpan(
+              text: ' !${community.name}',
+              style: TextStyle(color: theme.accentColor),
+              recognizer: TapGestureRecognizer()
+                ..onTap = () => goToCommunity.byId(
+                      context,
+                      community.instanceHost,
+                      community.id,
+                    ),
+            ),
+          ],
+        );
+
+    final modlogEntries = [
+      for (final removedPost in modlog.removedPosts)
+        _ModlogEntry.fromModRemovePostView(
+          removedPost,
+          RichText(
+            text: TextSpan(
+              children: [
+                if (removedPost.modRemovePost.removed)
+                  const TextSpan(text: 'removed')
+                else
+                  const TextSpan(text: 'restored'),
+                const TextSpan(text: ' post '),
+                TextSpan(
+                  text: '"${removedPost.post.name}"',
+                  style: TextStyle(color: theme.accentColor),
+                  recognizer: TapGestureRecognizer()
+                    ..onTap = () => goToPost(
+                          context,
+                          removedPost.instanceHost,
+                          removedPost.post.id,
+                        ),
+                ),
+              ],
+              style: TextStyle(color: theme.colorScheme.onSurface),
+            ),
+          ),
+        ),
+      for (final lockedPost in modlog.lockedPosts)
+        _ModlogEntry.fromModLockPostView(
+          lockedPost,
+          RichText(
+            text: TextSpan(
+              children: [
+                if (lockedPost.modLockPost.locked)
+                  const TextSpan(text: 'locked')
+                else
+                  const TextSpan(text: 'unlocked'),
+                const TextSpan(text: ' post '),
+                TextSpan(
+                  text: '"${lockedPost.post.name}"',
+                  style: TextStyle(color: theme.accentColor),
+                  recognizer: TapGestureRecognizer()
```

Comment:
is there any benefit of using `TapGestureRecognizer` over wrapping avatar + text in `InkWell`? I think generally we should move towards InkWells since it makes a cool ripple. (remember to use `borderRadius` so that it doesn't end up looking worse)

(also im not saying that inkwell is the best for everything, e.g. comment content that's clickable shouldn't probably be an inkwell)

Reply:
> is there any benefit of using TapGestureRecognizer over wrapping avatar + text in InkWell?

It doesn't add an ugly ripple

> I think generally we should move towards InkWells since it makes a cool ripple

ah

I'm not sure if ripples look good on link-type text but I'll check